docs/development.md

```diff
@@ -108,6 +112,59 @@ manifests creating a target cluster in AWS. After this is finished you will have
 a kubeconfig copied locally. You can debug most issues by SSHing into the
 instances that have been created and reading `/var/log/cloud-init-output.log`.
 
+#### Executing unit tests
+
+`make test` executes the project's unit tests. These tests do not stand up a
+Kubernetes cluster, nor do they have external dependencies.
+
+#### Executing integration tests
+`make integration` executes the project's integration tests. These tests stand
+up a local Kubernetes cluster using Kind in order to deploy the project's CRDs.
+The tested controller is **not** used to deploy Kuberentes to AWS.
+
+These tests depend on the following binaries in the system path:
+* `kind`
+
+#### Executing e2e tests
+`make e2e` executes the project's end-to-end tests with AWS account
+information parsed from the environment.
+
+These tests stand up a local Kubernetes cluster using Kind. The project's CRDs 
+and controllers are deployed to the Kind cluster and are used to deploy
+Kubernetes to AWS.
+
+The AWS janitor is by default configured to run after starting the end-to-end
+tests with the Makefile target. Please keep in mind that the janitor is highly
+destructive and should not be executed against shared AWS accounts or
+preferrably AWS accounts not dedicated to testing this project.
+
+`JANITOR_ENABLED=0 make e2e` executes the project's end-to-end tests with the
```

Comment:
Should this logic be inverted so that users don't mistakenly run the janitor?

Alternatively, is there a way to invoke the janitor through test-infra rather than here?

Reply:
Hi @detiber,

All of the documentation you see is for running the e2e tests from a developer desktop. The janitor in this case is the `aws-janitor` command, **not** the Boskos variant. This is simply trying to automate the resource cleanup step for developers running e2e locally. Remotely `JANITOR_ENABLED=0` will be set and thus the `aws-janitor` will **not** be run at the end of the e2e test execution, but rather the Boskos janitor running on the Prow cluster will be responsible for cleaning up resources.